### ISSUE TITLE
callback consistency

### ISSUE THREAD

**Issue by lukeolson** [pyamg/pyamg#186]
In the Krylov methods

``` python
        # Allow user access to residual
        if callback is not None:
            callback(normr)
        if keep_r:
            residuals.append(normr)
```

however in pyamg, the callback works on `xk`.  Using `xk` is more useful.  I propose changing them all to the current iterate...  thoughts?

**Comment by lukeolson** [comment]
checkout #187

**Comment by lukeolson** [comment]
fixed in de4bf95
